cli/ops/native_plugins.rs

```diff
@@ -0,0 +1,103 @@
+use super::dispatch_json::{Deserialize, JsonOp, Value};
+use crate::fs as deno_fs;
+use crate::ops::json_op;
+use crate::state::ThreadSafeState;
+use deno::*;
+use dlopen::symbor::Library;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+pub fn init(
+  i: &mut Isolate,
+  s: &ThreadSafeState,
+  isolate_arc: Arc<Mutex<deno::Isolate>>,
+) {
+  let isolate_arc_ = isolate_arc.clone();
+  i.register_op(
+    "open_native_plugin",
+    s.core_op(json_op(s.stateful_op(move |state, args, zero_copy| {
+      op_open_native_plugin(&isolate_arc_, state, args, zero_copy)
+    }))),
+  );
+}
+
+fn open_plugin<P: AsRef<OsStr>>(lib_path: P) -> Result<Library, ErrBox> {
+  debug!("Loading Native Plugin: {:#?}", lib_path.as_ref());
+
+  Library::open(lib_path).map_err(ErrBox::from)
+}
+
+struct NativePluginResource {
+  lib: Library,
+  ops: Vec<(String, OpId)>,
+}
+
+impl Resource for NativePluginResource {}
+
+struct InitContext {
+  isolate: Arc<Mutex<deno::Isolate>>,
```

Comment:
I don't like that isolate is passed to native plugin, IMHO users shouldn't be able to control CLI isolate. Is it possible that plugin init function returns vector of (`name`, `boxed op`) tuples?

Reply:
None of the fields in `InitContext` are directly exposed to the plugin. It might be possible to do it unsafely.